core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -229,6 +253,11 @@ public void initialize(String name, Map<String, String> unresolved) {
     // build the final configuration and set up the catalog's auth
     Map<String, String> mergedProps = config.merge(props);
     Map<String, String> baseHeaders = configHeaders(mergedProps);
+    this.endpoints = ImmutableSet.copyOf(config.endpoints());
+    if (endpoints.isEmpty()) {
```

Comment:
Nit: why not check if `config.endpoints()` is present and set `this.endpoints` just once?